source/device.hpp

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <set>
+#include <unordered_map>
+
+namespace sim {
+
+class Link;
+
+enum DeviceType { SWITCH, SENDER, RECEIVER };
+
+class Device {
+public:
+    Device(DeviceType a_type);
+    virtual ~Device();
+    virtual void process();
+    void add_inlink(Link *link);
+    void add_outlink(Device *dest, Link *link);
```

Comment:
Link already contains information about destination device
```suggestion
    void add_outlink(Link *link);
```

Reply:
See comment above: https://github.com/Cloud-Storage-Team/algnet/pull/49#discussion_r1959740648

Reply:
Ok, understood, thank you. But naming is very confused now, I think, it
```suggestion
    // updates routing table: sets given Link as next on the path to given Device
    void add_route(Device *dest, Link *link);
```